src/x/serialize/decoder_test.go

```diff
@@ -179,6 +182,26 @@ func TestDecodeLiteralTooLong(t *testing.T) {
 	require.Error(t, d.Err())
 }
 
+func TestDecodeLiteralOfMaximumPossibleLength(t *testing.T) {
+	name := strings.Repeat("n", math.MaxUint16)
+	value := strings.Repeat("v", math.MaxUint16)
+	b := bytes.Join([][]byte{
+		headerMagicBytes,
+		encodeUInt16(1, make([]byte, 2)),              // num tags
+		encodeUInt16(math.MaxUint16, make([]byte, 2)), // name length
+		[]byte(name),
+		encodeUInt16(math.MaxUint16, make([]byte, 2)), // value length
+		[]byte(value),
+	}, nil)
```

Comment:
Why not use the encoder for this, also testing both encoder and decoder at the same time?

Reply:
Tried to keep it consistent with other tests, since all the tests in this file start with byte array and then decode it.

Added a separate round-trip test in https://github.com/m3db/m3/pull/3815/commits/56ef8055f981289be6a90d616b7f2faac958f659 . Let me know if I should move it somewhere else aand if I should remove some of the tests in this PR.